src/Games/NexusMods.Games.MountAndBlade2Bannerlord/MountAndBlade2BannerlordLoadoutSynchronizer.cs

```diff
@@ -0,0 +1,88 @@
+using Microsoft.Extensions.DependencyInjection;
+using NexusMods.Abstractions.GameLocators;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.Abstractions.Loadouts.Synchronizers;
+using NexusMods.Abstractions.Settings;
+using NexusMods.Extensions.BCL;
+using NexusMods.Games.MountAndBlade2Bannerlord.Models;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.Paths;
+using NexusMods.Paths.Extensions;
+using static NexusMods.Games.MountAndBlade2Bannerlord.MountAndBlade2BannerlordConstants;
+
+namespace NexusMods.Games.MountAndBlade2Bannerlord;
+
+public class MountAndBlade2BannerlordLoadoutSynchronizer : ALoadoutSynchronizer
+{
+    public MountAndBlade2BannerlordLoadoutSynchronizer(IServiceProvider provider) : base(provider)
+    {
+        var settingsManager = provider.GetRequiredService<ISettingsManager>();
+        _settings = settingsManager.Get<MountAndBlade2BannerlordSettings>();
+    }
+
+    private readonly MountAndBlade2BannerlordSettings _settings;
+
+    public override bool IsIgnoredBackupPath(GamePath path)
+    {
+        if (_settings.DoFullGameBackup) return false;
+        return true;
+    }
+
+    protected override ValueTask MoveNewFilesToMods(Loadout.ReadOnly loadout, IEnumerable<AddedEntry> newFiles, ITransaction tx)
+    {
+        var modDirectoryNameToModel = new Dictionary<RelativePath, ModLoadoutItem.ReadOnly>();
+
+        foreach (var newFile in newFiles)
+        {
+            GamePath gamePath;
+
+            if (!IsModFile(newFile.LoadoutItemWithTargetPath.TargetPath, out var modDirectoryName))
+            {
+                continue;
+            }
+
+            if (!modDirectoryNameToModel.TryGetValue(modDirectoryName, out var mod))
+            {
+                if (!TryGetSMAPIMod(modDirectoryName, loadout, loadout.Db, out mod))
+                {
+                    continue;
+                }
+
+                modDirectoryNameToModel[modDirectoryName] = mod;
+            }
+            
+            newFile.LoadoutItem.ParentId = mod.Id;
+        }
+        return ValueTask.CompletedTask;
+    }
+
+    private static bool TryGetSMAPIMod(RelativePath modDirectoryName, Loadout.ReadOnly loadout, IDb db, out ModLoadoutItem.ReadOnly mod)
```

Comment:
This function needs renamed; since it was copied from Stardew.

-------

On a misc (unrelated) note, I had a quick walk through the codebase, because I was curious if there's any potential issues with file casings. There shouldn't be in this case, but if you change the casing of an existing game folder on Linux, I imagine stuff would break; because the App would use the casing on original file path as source of truth.